### PR TITLE
Fix #4906 - warn on potentially conflicting ACMG terms.

### LINKED ISSUE #4906: ACMG criteria that cannot be used together
**Is your feature request related to a problem in the current program to new available techology or software? Please describe and add links/citations if appropriate.**

Certain ACMG criteria are not allowed together. In general one evidence should only count towards one term (eg PVS1 not together with PM4, which can be easily missed by the operator).

**Describe the solution you'd like**
Warn / flag when users fill in an incompatible combo.

**Describe alternatives you've considered**
It would be possible to forbid illegal ones. It is probably more flexible to just warn - eg gene specific criteria may say otherwise.

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality or fixes a bug.

- Fix #4906 

Thanks to @4WGH for highlighting this and for a good discussion on potential pairs.

Screenshot 2024-10-10 at 16 44 39

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

**How to test**:
1. how to test it, possibly with real cases/data

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

File: scout/constants/acmg.py [hunk 1]
```diff
@@ -297,7 +297,7 @@
                     (
                         "BP7",
                         {
-                            "short": "(not in use)",
+                            "short": "Synonymous, no splice impact",
```
Comment (dnil) [inline scout/constants/acmg.py:300]: I’m going to guess that was from before we had splice prediction in the pipe.

Comment (northwestwitch) [inline scout/constants/acmg.py:300]:
```suggestion
                            "short": "Synonymous, no impact on splicing",
```

File: scout/utils/acmg.py [hunk 2]
```diff
@@ -1,4 +1,9 @@
 # coding=UTF-8
+from flask import flash
```
Comment (northwestwitch) [inline scout/utils/acmg.py:2]:
```suggestion
```

**Review by northwestwitch — APPROVED** [review]
Really nice new feature. Very visible and useful! 🥇 
 I have a couple of minor suggestions in the code.

File: scout/utils/acmg.py (hunk 2 above)
Comment (dnil) [reply scout/utils/acmg.py]: Ahem, right - decided on the js path instead. 👍

**Pull request merged**